ssh-agent-canonicalize

```diff
@@ -15,22 +24,22 @@ function canon_agent_pid() {
     source "$ENV_FILE" >/dev/null
     # If the auth socket file is gone then the agent is unreachable. Even if it
     # was canonical at some point, there's no reason to keep it around any more.
-    if [ -S "$SSH_AUTH_SOCK" ]; then
+    if [[ -S "${SSH_AUTH_SOCK:-}" ]]; then
       echo ${SSH_AGENT_PID:-}
     fi
   )
 }
 
 function non_canon_agent_pids() {
-  echo $(pgrep ssh-agent) | sed -e "s#\b$(canon_agent_pid)\b##"
+  echo $( pgrep ssh-agent | grep -v "^$(canon_agent_pid)$" )
```

Comment:
I don't remember the history of why I made this function so complicated, but we can actually go one step further and remove the echo:
```
pgrep ssh-agent | grep -v "^$(canon_agent_pid)$"
```